### PR TITLE
Feature undo redo

### PR DESCRIPTION
**Pull request by Dartemiss**
Refactor Undo/Redo:
- All logic moved to ModuleActions
- Now actions that adds/deletes gameobjects or components are serialized in memory and load, it is instantly

### CODE REVIEW

File: Engine/Actions/EditorActionSetTexture.h [hunk 1]
```diff
@@ -8,14 +8,14 @@ class EditorActionSetTexture : public EditorAction
...
-	~EditorActionSetTexture();
+	~EditorActionSetTexture() = default;
 
 
 	void Undo();
 	void Redo();
 
 private:
-	ComponentMaterial* component;
+	uint64_t UUID_COMP;
```
Comment (mariofv) [inline Engine/Actions/EditorActionSetTexture.h]: Naming should be `component_UUID` or `UUID_component` in order to match naming style.

File: Engine/Actions/EditorActionModifyLight.h [hunk 2]
```diff
@@ -14,9 +14,9 @@ class EditorActionModifyLight : public EditorAction
 	void Redo();
 
 private:
-	ComponentLight* component;
 	float color[3];
 	float intensity;
+	uint64_t UUID_COMP;
```
Comment (mariofv) [inline Engine/Actions/EditorActionModifyLight.h]: Same as previous comment

File: Engine/Actions/EditorActionAddComponent.h [hunk 3]
```diff
@@ -7,14 +7,17 @@
...
+	EditorActionAddComponent(Component* comp);
+	~EditorActionAddComponent() = default;
 
 	void Undo();
 	void Redo();
 
 private:
-	Component* component;
+	uint64_t UUID_COMP;
+	uint64_t UUID_OWNER;
```
Comment (mariofv) [inline Engine/Actions/EditorActionAddComponent.h]: Same as previous comments 😄

**Review by mariofv — CHANGES_REQUESTED** [review]
Great job, it now looks clearer :D Just change the naming as commented and it will be good to go

File: Engine/Actions/EditorActionSetTexture.h (hunk 1 above)
Comment (Dartemiss) [reply Engine/Actions/EditorActionSetTexture.h]: Okay, I just have change it and pushed it. :D

**Review by mariofv — APPROVED** [review]

**Review by yalania — APPROVED** [review]
Looks good :) You can change most of the EditorAction's constructors to use member initialization if you want. It is really good explained here https://www.learncpp.com/cpp-tutorial/8-5a-constructor-member-initializer-lists/

**Pull request merged**